index.ts

```diff
@@ -420,10 +453,31 @@ class IntegrationAPI extends EventEmitter {
           this.emit(api.Events.SetupDriverAbort);
           break;
 
+        case api.MsgEvents.Oauth2Authorization:
+          this.emit(api.Events.Oauth2Authorization, msgData);
+          break;
+
+        case api.MsgEvents.Oauth2Refreshed:
+          this.emit(api.Events.Oauth2Refreshed, msgData);
+          break;
+
         default:
           log.warn(`[${wsId}] Unhandled event: ${msg}`);
           break;
       }
+    } else if (kind === "resp") {
+      const statusCode = json.code;
+      const pendingRequest = this.#pendingRequests.get(json.req_id);
+      if (!pendingRequest) {
+        log.warn(`[${wsId}] No pending request found for id ${id}`);
+        return;
+      }
+      if (statusCode == 200) {
```

Comment:
The status code corresponds to the http status codes: https://github.com/unfoldedcircle/core-api/blob/ee10ae69784fad38526fe466ff0e874db0369e73/integration-api/UCR-integration-asyncapi.yaml#L828-L831

Therefore, any 2xx code is a success code.
Please change valid code to range 200-299